source/docs/software/networktables/creating-sendable-classes.rst

```diff
@@ -0,0 +1,124 @@
+
+Creating a new Sendable class with SendableBuilder
+--------------------------------------------------
+
+To expose information about a class to users over NetworkTables, one must create an implementation of the initSendable() method of the Sendable interface. The properties added to Sendable classes can include just a getter, or both a getter and a setter. This example shows a Double property which can both be viewed and modified on NetworkTables. Note that both a getter and setter are not required - a :code:`null` setter simply means the value can't be changed over NetworkTables. This might be desired in some situations where information should only be displayed, but not modified. If the Sendable is configured as an actuator, the setter will be called when the robot is enabled in Test mode when users change it's values to test robot functions.
+
+.. tabs::
+
+    .. code-tab:: c++
+
+        void DifferentialDrive::InitSendable(SendableBuilder& builder) {
+
+            builder.AddDoubleProperty("Intake Speed",
+                                    [=]() { return
+                                    intakeMotor.Get(); },
+                                    [=](double value) {
+                                    intakeMotor.Set(value);
+                                    });
+
+        }
+
+    .. code-tab:: java
+
+        @Override
+        public void initSendable(SendableBuilder builder) {
+
+            builder.addDoubleProperty("Intake Speed",
+                intakeMotor::get, intakeMotor::set);
+
+        }
+
+The following example is an example implementation from WPILib's DifferentialDrive class, which implements Sendable. The properties added to the builder in this example expose many features of DifferentialDrive to modification through NetworkTables. The instance of DifferentialDrive is treated as an actuator of type "DifferentialDrive", which means that Test mode can be used to control the drive's outputs, and the name DifferentialDrive will be displayed to the user. When Test mode is enabled or disabled, the actuator will be set to a safe state by calling the :code:`stopMotor` method, which will stop the motors. Finally, a getter and setter for the left motor speed and right motor speed allows the user both to view the current output of both motors, as well as set them to an arbitrary output. For more information on the DifferentialDrive class, see :ref:`wpi_differential_drive`.
+
+.. tabs::
+
+    .. code-tab:: c++
+
+        void DifferentialDrive::InitSendable(SendableBuilder& builder) {
+            builder.SetSmartDashboardType("DifferentialDrive");
+            builder.SetActuator(true);
+            builder.SetSafeState([=] { StopMotor(); });
+            builder.AddDoubleProperty("Left Motor Speed",
```

Comment:
This formatting is bizarre. Should be

```cpp
builder.AddDoubleProperty("Left Motor Speed",
                          [=]() { return m_leftMotor.Get(); },
                          [=](double value) { m_leftMotor.Set(value); });
builder.AddDoubleProperty(
    "Right Motor Speed",
    [=]() { return m_rightMotor.Get() * m_rightSideInvertMultiplier; },
    [=](double value) {
      m_rightMotor.Set(value * m_rightSideInvertMultiplier);
    });
```

Don't split lambda functions into separate lines like that, it's very difficult to read